### PR TITLE
Feature/327: implement jupyter node selectors

### LINKED ISSUE #327: Jupyterhub enable node selector per application
Closes #327

### PR DESCRIPTION
**Pull request by zsinnema**
Closes #327

### CODE REVIEW

**Comment by zsinnema** [comment]
example:
```
harness:
  jupyterhub:
    node_selector:             # optional node pool for these pods
      - key: ch/nodepool
        operator: In           # In, NotIn, Exists, DoesNotExist. Gt, and Lt.
        values: pool-jupyter
        matchPurpose: require  # require | prefer | ignore
```

File: applications/jupyterhub/src/harness_jupyter/harness_jupyter/jupyterhub.py [hunk 1]
```diff
@@ -45,9 +45,41 @@ def change_pod_manifest(self: KubeSpawner):
...
+                    #       node_selector:             # optional node pool for these pods
+                    #         - key: ch/nodepool
+                    #           operator: In           # In, NotIn, Exists, DoesNotExist. Gt, and Lt.
+                    #           values: pool-jupyter
+                    #           matchPurpose: require  # require | prefer | ignore
+                    spawnerExtraConfig = harness['jupyterhub'].get('spawnerExtraConfig')
+                    if spawnerExtraConfig:
+                        for k, v in spawnerExtraConfig.items():
                             setattr(self, k, v)
+                        node_selectors = spawnerExtraConfig.get('node_selectors')
```
Comment (filippomc) [inline applications/jupyterhub/src/harness_jupyter/harness_jupyter/jupyterhub.py:57]: here is `node_selectors`, the doc above says `node_selector`. Should also be in camelCase to match the general k8s spec convention. Need to also document it in the jupyterhub application readme

**Review by filippomc — CHANGES_REQUESTED** [review]
Need to fix a conflict and a mismatch between the doc strings and the name of the variable.
I also suggest to add some docs about how to configure this in the jupyterhub application README

File: applications/jupyterhub/src/harness_jupyter/harness_jupyter/jupyterhub.py (hunk 1 above)
Comment (zsinnema) [reply applications/jupyterhub/src/harness_jupyter/harness_jupyter/jupyterhub.py:57]: fixed the type, thanks for catching

Comment (zsinnema) [reply applications/jupyterhub/src/harness_jupyter/harness_jupyter/jupyterhub.py:57]: @filippomc about the camelCase: the other jupyter configs aren't camelCase e.g. `cpu_guarantee`

**Review by filippomc — APPROVED** [review]

**Pull request merged**